lib/mwUtil.js

```diff
@@ -386,4 +386,54 @@ mwUtil.isSelfRedirect = (req, location) => {
     return true;
 };
 
+/**
+ * Traverses through a request body and replaces the given keys with
+ * the content fetched using the 'fetch' callback
+ *
+ * @param responce
+ * @param fetch
+ * @param mergeKey
+ */
+mwUtil.populate = (responce, fetch, mergeKey) => {
```

Comment:
Alternative naming ideas: 

- `hydrateResources`
- `hydrateJSON`

My main gripe with `populate` is that it's fairly generic. Hydration is more commonly used for the process of expanding ids / references to actual content, for example in the context of a twitter feed.